### PR TITLE
liuliu/webhook-doc

### PR DESCRIPTION
**Pull request by liuliu-dev**

### CODE REVIEW

File: content/guides/dolthub-api/hooks.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+DoltHub hooks currently only support push, pull request and branch events on a database, while GitHub supports a
+much [longer list of
+events](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads)
+on both [organizations](https://docs.github.com/en/rest/orgs/webhooks) and
+[repositories](https://docs.github.com/en/rest/repos#webhooks).
+
+If you have an event you'd like us to support, [file an issue](https://github.com/dolthub/dolthub-issues/issues/new?assignees=tbantle22&labels=enhancement&template=feature_request.md&title=) or reach out to us on [Discord](https://discord.com/invite/RFwfYpu).
+
+## Set up webhooks
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]:
```suggestion
## Setting up webhooks: an example
```

File: content/guides/dolthub-api/hooks.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Events and payloads
+
+### on_push
+
+This event occurs when a commit is pushed.
+
+#### Webhook payload object
+
+The payload delivered for a push event webhook contains information about the push that was made to the repository. The payload is a JSON object that includes the following fields:
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]: repository -> database

File: content/guides/dolthub-api/hooks.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  "repository": {
+    "name": "test",
+    "owner": "liuliu"
+  }
+}
+```
+
+### on_pull_request
+
+The Pull Request event webhook is triggered whenever a pull request is created, merged, or closed, the payload includes these fields:
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]:
```suggestion
The pull request event webhook is triggered whenever a pull request is created, merged, or closed. The payload includes these fields:
```

File: content/guides/dolthub-api/hooks.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+The payload delivered for a push event webhook contains information about the push that was made to the repository. The payload is a JSON object that includes the following fields:
+
+| Key                  | Type    | Description                                                                    |
+| ---------------------| --------| -------------------------------------------------------------------------------|
+| `ref`                | string  | The name of the branch that was pushed.                                        |
+| `head`               | string  | The SHA of the most recent commit on `ref` after the push.                     |
+| `prev`               | string  | The SHA of the most recent commit on `ref` before the push.                    |
+| `repository.name`    | string  | The name of the repository where the push occurred.                            |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.           |
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]: Repository owners can be orgs too. Applies to all sections
```suggestion
| `repository.owner`   | string  | The owner (username or org name) of the repository where the push occurred.           |
```

File: content/guides/dolthub-api/hooks.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+### on_pull_request
+
+The Pull Request event webhook is triggered whenever a pull request is created, merged, or closed, the payload includes these fields:
+
+| Key                  | Type    | Description                                                                   |
+| ---------------------| --------| ------------------------------------------------------------------------------|
+| `repository.name`    | string  | The name of the repository where the push occurred.                           |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.          |
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]: "where the push occurred" -> "where the pull request change occurred"?

File: content/guides/dolthub-api/hooks.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    "owner": "liuliu"
+  },
+  "pullID": "15",
+  "action": "Opened"
+}
+```
+
+### on_branch
+
+A Branch Event Webhook is triggered when a branch is created or deleted, the payload includes these fields:
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]:
```suggestion
A branch event webhook is triggered when a branch is created or deleted. The payload includes these fields:
```

File: content/guides/dolthub-api/hooks.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+| Key                  | Type    | Description                                                                   |
+| ---------------------| --------| ------------------------------------------------------------------------------|
+| `repository.name`    | string  | The name of the repository where the push occurred.                           |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.          |
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md]: "where the push occurred" -> "where the branch change occurred"

File: content/guides/dolthub-api/hooks.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+| `event_name	`        | string  | The name of the event (branch).                                               |
+| `branch`             | string  | The full name of the branch (e.g., refs/heads/main).                          |
+| `action`             | string  | The action that triggered the webhook (deleted or created the branch).        |
+
+An example of the payload:
+```json
+{
+  "event_name": "branch",
```
Comment (tbantle22) [inline content/guides/dolthub-api/hooks.md:111]: Why does this have an event name and the others don't?

**Review by tbantle22 — APPROVED** [review]

File: content/guides/dolthub-api/hooks.md (hunk 8 above)
Comment (liuliu-dev) [reply content/guides/dolthub-api/hooks.md:111]: i added this `event_name` to the branch payload in dolthubapi, I didn't realize the others don't have it, and didn't want to change the existing ones to affect customer use..

**Pull request merged**